### DISCUSSION TITLE
etcd tuning for high latency?

### DISCUSSION THREAD

**Discussion by justyns** [siderolabs/talos#8914] [Q&A] [answered]
Hello!

Does Talos do any special tuning to etcd to handle high latency?   AFAIK the usual recommendations to not let k8s clusters span multiple regions/datacenters is due to etcd not liking high latency by default.  

I've had luck in the past setting the parameters mentioned on https://etcd.io/docs/v3.4/tuning/ (heartbeat-interval and election-timeout).

Is this something Talos already handles, or is there a way to set it for etcd?

**Comment by rsmitty** [comment, marked as answer]
It's not something we do by default, but you can certainly add those tuning parameters yourself with extraArgs in the etcd section of the machine config: https://www.talos.dev/v1.7/reference/configuration/v1alpha1/config/#Config.cluster.etcd

**Reply by justyns** [reply]
Thanks!  I looked for something similar but must have missed it.  I created a new patch that does what I want:

``` yaml
cluster:
  etcd:
    extraArgs:
      election-timeout: "50000"
      heartbeat-interval: "5000"
```
